nautobot/core/models/managers.py

```diff
@@ -13,6 +16,33 @@ class BaseManager(Manager):
     Adds built-in natural key support, loosely based on `django-natural-keys`.
     """
 
+    def bulk_create(
+        self,
+        objs,
+        batch_size=None,
+        ignore_conflicts=False,
+        change_log=False
+    ):
+        if not change_log:
+            return super().bulk_create(objs, batch_size, ignore_conflicts)
+        from nautobot.extras.choices import ObjectChangeActionChoices
+        from nautobot.extras.models import ObjectChange, ChangeLoggedModel
+        with transaction.atomic():
+            created_objects = super().bulk_create(
+                objs=objs,
+                batch_size=batch_size,
+                ignore_conflicts=ignore_conflicts,
+            )
+            object_changes = []
+            for obj in created_objects:
+                if not isinstance(obj, ChangeLoggedModel):
+                    continue
+                object_change = obj.to_objectchange(action=ObjectChangeActionChoices.ACTION_CREATE)
+                object_change.request_id = obj.pk  # Hack, should probably be a common id for all the objects
```

Comment:
Yeah, at minimum this should be a common id within this function call; consider maybe also letting the caller optionally pass a `request_id` and/or `user` as additional kwargs to the function?

Reply:
I like all of those ideas!